src/w_wad.c

```diff
@@ -149,7 +149,7 @@ static void W_AddFile(const char *name) // killough 1/31/98: static, const
       lseek(handle, header.infotableofs, SEEK_SET);
       // [FG] check return value
       if (!read(handle, fileinfo, length))
-        I_Error("Error reading lump directory from %s\n", filename);
+        I_Printf(VB_WARNING, "Error reading lump directory from %s\n", filename);
```

Comment:
Shouldn't we close the file, free the file name and return in that case?

Reply:
No, if we exit this function early, we will leave NULL.WAD in the user's tmp directory.

Perhaps we should rewrite this function with proper error checking and cleanup on exit. For example, `malloc(header.numlumps * sizeof(filelump_t))` may be too large, check `lseek`, etc.

Reply:
But then we should probably nullify the header, so the loops that follow get short-cut?

Reply:
I think we should check everything or just leave it as it is. Why all these checks if, for example, `lseek` one line earlier is not checked?

Reply:
I think to fix the immediate issue we should set `header.numlumps = 0` to invalidate the header.